index.js

```diff
@@ -7,12 +7,105 @@ function ask(questionText) {
   });
 }
 
+
 start();
 
 async function start() {
-  console.log("Let's play a game where you (human) make up a number and I (computer) try to guess it.")
-  let secretNumber = await ask("What is your secret number?\nI won't peek, I promise...\n");
+  console.log("\nLet's play a game where you make up a number and I try to guess it.")
+  let secretNumber = await ask("\nWhat is your secret number?\nI won't peek, I promise...\n");
   console.log('You entered: ' + secretNumber);
-  // Now try and complete the program.
-  process.exit();
+  //secretNumber = the number user inputs for computer to guess.
+  function compGuess(max, min) {
+    //compGuess = Function for Computer generated guess
+    return (min) + Math.floor(Math.random() * (max - min))
+
+  }
+  game()
+
+  async function game() {
+    let max = Number(process.argv[2]) || 100
+    //allows user to input a mx number for the guessing range. If no number inputed will revert to 100
+    let min = 1
+    let turns = 0;
+    //turns is the amount of tries it takes the computer to guess my number
+    console.log("\nOkay computer give it your best shot!\nGuess a number between 1 and " + max + "!")
+    let guess = compGuess(max, min)
+    // takes compGuess function and assigns it to a variable.
+    let playerAnswer = await ask("Is your number " + guess + "? y/n ")
+    // player answer is the players response to the computers guess. in the form of y / n
+    if (guess > secretNumber) {
+      console.log("Too High! Guess again!\n")
+      max = guess
+      // reassigning the max to the guess allows for the guessing range to be narrowed down.
+    } else if (guess < secretNumber) {
+      console.log("Too low! Guess again!\n")
+      min = guess + 1
+    }
+    while (playerAnswer !== "yes" || playerAnswer !== "y") {
+      guess = compGuess(max, min)
+      playerAnswer = await ask("Is your number " + guess + "? y/n ")
+
+      turns = turns + 1
```

Comment:
Here you increment turns, but at this point during the first run through the loop, the computer has actually guessed twice. You will want to increment turns every time the computer asks if a number is correct, like on lines 34 and 46.